File: app/src/main/java/fr/xgouchet/chronorg/ui/contracts/presenters/BasePresenter.java
package fr.xgouchet.chronorg.ui.contracts.presenters;

import android.support.annotation.NonNull;

/**
 * @author Xavier Gouchet
 */
public interface BasePresenter<V, T> {

    void setView(@NonNull V view);

    void subscribe();

    void unsubscribe();

    void load(boolean force);
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/presenters/BasePresenter.kt
package fr.xgouchet.khronorg.ui.presenters

import fr.xgouchet.khronorg.ui.views.View
import kotlin.properties.Delegates.notNull

/**
 * @author Xavier F. Gouchet
 */
class BasePresenter<T : Any> : Presenter<T> {

    var view: View<T> by notNull()

    var data: T? = null

    override fun subscribe() {
        data?.let { view.setContent(it) }
    }

    override fun unsubscribe() {
        // nothing to do
    }

    override fun load(force: Boolean) {
        // nothing to load
    }

}